### PR TITLE
Wounds UI class and components

### PR DESCRIPTION
**Pull request by Lemon-King**
Class and Components for Character, FriendlyTarget, and HostileTarget.

### CODE REVIEW

**Comment by codecorsair** [comment]
Your branch has conflicts and can not be automatically merged.  Please resolve them.

File: src/ts/components/woundframe/components/Stat.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import * as React from 'react';
+
+export interface StatState { }
+export interface StatProps {
+  type: string,
+  value: number,
+  max: number
+}
+
+export class Stat extends React.Component<StatProps, StatState> {
```
Comment (codecorsair) [inline src/ts/components/woundframe/components/Stat.tsx]: define a default export to export the Stat class as well.

File: src/ts/components/woundframe/WoundFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import { Doll } from './components/Doll';
+import { Stats } from './components/Stats';
+import { Buffs } from './components/Buffs';
+import { WoundColors } from './classes/WoundColors';
```
Comment (codecorsair) [inline src/ts/components/woundframe/WoundFrame.tsx]: When you're importing a default import you do not need to use object destructuring. 

These all give you the same result without the extra characters.
```javascript
import Doll from './components/Doll';
import Stats from './components/Stats';
import Buffs from './components/Buffs';
import WoundColors from './classes/WoundColors';
```

File: src/ts/components/woundframe/WoundFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  injuries: any[];
+  health: number;
+  healthMax: number;
+  stamina: number;
+  staminaMax: number;
+  //panic: number;
+  //panicMax: number;
+  //temp: number;
+  //tempMax: number;
+  //colors: any;
```
Comment (codecorsair) [inline src/ts/components/woundframe/WoundFrame.tsx]: Shouldn't these be used?  I assume you have these commented out in order to use them in the future?  If you're mocking data do it at the highest level, which would be when you consume the WoundFrame component so you do not have to come back and edit the WoundFrame AND the modules consuming it in the future but only the consumers.

File: src/ts/components/woundframe/classes/WoundColors.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    const value = localStorage.getItem(this.key);
+    if (value && typeof value === "string") {
+      this.colors = value.split(",");
+
+      // stored color validation
+      for (var i = 0; i < this.colors.length; i++) {
+        const color: string = this.colors[i];
+        if (color.length !== 7 && color.substring(0,0) !== "#") {
+          this.colors[i] = "#ffffff";
+        }
```
Comment (codecorsair) [inline src/ts/components/woundframe/classes/WoundColors.tsx]: This check can still pass and be an invalid value.

Using a regex you can get a much more accurate check. From [StackOverflow](http://stackoverflow.com/questions/8027423/how-to-check-if-a-string-is-a-valid-hex-color-representation)
```javascript
var isOk  = /(^#[0-9A-F]{6}$)|(^#[0-9A-F]{3}$)/i.test('#ac3');
```

File: src/ts/components/woundframe/components/Stat.tsx (hunk 1 above)
Comment (Lemon-King) [inline src/ts/components/woundframe/components/Stat.tsx]: Fixed

File: src/ts/components/woundframe/WoundFrame.tsx (hunk 2 above)
Comment (Lemon-King) [inline src/ts/components/woundframe/WoundFrame.tsx]: Fixed and altered WoundColors to import correctly.

File: src/ts/components/woundframe/WoundFrame.tsx (hunk 3 above)
Comment (Lemon-King) [inline src/ts/components/woundframe/WoundFrame.tsx]: Used now and integrated into frames.

File: src/ts/components/woundframe/classes/WoundColors.tsx (hunk 4 above)
Comment (Lemon-King) [inline src/ts/components/woundframe/classes/WoundColors.tsx]: Added and updated the check.

**Pull request merged**